core/src/main/scala/scala/pickling/pickler/Nothing.scala

```diff
@@ -0,0 +1,16 @@
+package scala.pickling.pickler
+
+import scala.pickling._
+import PicklingErrors.NothingIsNotUnpicklable
+
+trait NothingPicklers {
+  implicit object NothingPicklerUnpickler extends AbstractPicklerUnpickler[Nothing] {
+    override def tag: FastTypeTag[Nothing] = FastTypeTag.Nothing
+    /** Impossible to call in Scala, no value can be typed [[Nothing]] */
+    override def pickle(picklee: Nothing, builder: PBuilder): Unit = ???
```

Comment:
I'd throw a "NothingIsNotPicklable" exception here, or possibly we can create a macro that fails with that error message at compile time.